## Soal Nomor 1
* Klasifikasi menggunakan Algoritma Naive Bayes
* Berapa persen tingkat pembelelian komputer dari Student = Yes dengan tingkat income = Low

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
data = pd.read_csv("C:/Users/USER/Desktop/PMM-DN-STIMATA MALANG/UAS/dataset_UAS-main/dataset_UAS-main/dataset_soal No.1.csv")

In [3]:
data.head()

Age  Income Student Credit_rating Clas(buy_computer)
0   <=30    High      No          Fair                 No
1   <=30    High      No     Excellent                 No
2  31.40    High      No          Fair                Yes
3    >40  Medium      No          Fair                Yes
4    >40     Low     Yes          Fair                Yes

In [4]:
data.shape

(51, 5)

In [5]:
data['Clas(buy_computer)'].value_counts()

Yes    29
No     22
Name: Clas(buy_computer), dtype: int64

In [6]:
data['Age'].value_counts()

<=30     20
>40      17
31.40    14
Name: Age, dtype: int64

In [7]:
data['Income'].value_counts()

Low       21
Medium    19
High      11
Name: Income, dtype: int64

In [8]:
data['Student'].value_counts()

Yes    27
No     24
Name: Student, dtype: int64

In [9]:
data['Credit_rating'].value_counts()

Fair         31
Excellent    20
Name: Credit_rating, dtype: int64

In [10]:
PYes = (29/51)
PNo = (22/51)

In [11]:
print(PYes)

0.5686274509803921


In [12]:
print(PYes)

0.5686274509803921


## Probabilitas Pertama
#### P(A|B) ?  Jika P(A) = Tidak membeli dan P(B) = Umur dengan kategori sedang (<=30 tahun)
P(A) = Clas(buy_computer) & P(B) = Age

In [13]:
pd.crosstab(data['Age'], data['Clas(buy_computer)'])

Clas(buy_computer)  No  Yes
Age                        
31.40                4   10
<=30                10   10
>40                  8    9

In [14]:
PKecilNo = 4/22
PSedangNo = 10/22
PBesarNo = 8/22

PKecilYes = 10/29
PSedangYes = 10/29
PBesarYes = 9/29

PKecil = 14/51
PSedang = 20/51
PBesar = 17/51

In [15]:
#Perhitungan probabilitas yang tidak membeli komputer dengan umur <=30 tahun
PNoSedang = (PSedangNo * PNo) / PSedang
print(PNoSedang)

0.5


In [16]:
##Perhitungan probabilitas yang membeli komputer dengan umur <=30 tahun
PYesSedang = (PSedangYes * PYes) / PSedang
print(PYesSedang)

0.5


### Conclusion 1
Probabilitas (Pembelian komputer | Umur <=30 tahun) sebanding dengan probalilitas (Pembelian komputer | Umur <=30 tahun)

In [17]:
print(PNoSedang)
print(PYesSedang)

0.5
0.5


## Probabilitas Kedua
#### P(A |  B) = P(Clas(buy_computer) | Age Sedang, Income Low, Student Yes)
Dimana
* P(A) adalah Clas(buy_computer) / yang membeli komputer
* P(B) adalah Age = sedang ; Income = Low ; Student = Yes

In [18]:
pd.crosstab(data['Income'], data['Clas(buy_computer)'])

Clas(buy_computer)  No  Yes
Income                     
High                 6    5
Low                 11   10
Medium               5   14

In [19]:
PHigIncomehNo = 6/22
PLowIncomeNo = 11/22
PMediumIncomeNo = 5/22

PHighIncomeYes = 5/29
PLowIncomeYes = 10/29
PMediumIncomeYes = 14/2

PHighIncome = 11/51
PLowIncome = 21/51
PMediumIncome = 19/51

In [20]:
pd.crosstab(data['Student'], data['Clas(buy_computer)'])

Clas(buy_computer)  No  Yes
Student                    
No                  10   14
Yes                 12   15

In [21]:
PNoStudentNo = 10/22
PYesStudentNo = 12/22

PNoStudentYes = 12/29
PYesStudentYes = 15/29

PNoStudent = 24/51
PYesStudent = 27/51

In [22]:
#P(A) = Clas(buy_computer)  = Buy = PYes

#Kondisi B Dimana
#Age = Sedang
#Income = Low
#Student = Yes
#P(B) = PSedang, PLow, PYesStudent

#P(B)1 = PSedang
#P(B)2 = PLow
#P(B)3 = PYesDStudent

#P(B|A)1 = (PSedang | Buy) = PSedangYes
#P(B|A)2 = (PLow | Buy) = PLowIncomeYes
#P(B|A)3 = (PYesStudent | Buy) = PYesStudentYes


#Yang ditanyakan P(A|B) ?
#Rumus
#P(A|B) = P(B|A) * P(A) / P(B)
#P(YesBuy) = ((PSedangYes * PLowIncomeYes * PYesStudentYes) * PYes) / (PSedangYes * PLowIncomeYes * PYesStudentYes)

In [23]:
#Probabilitas ClasYes dengan StudentYes, IncomeLow dan AgeKecil
PYesBuyA = ((PKecilYes * PLowIncomeYes * PYesStudentYes) * PYes) / (PKecilYes * PLowIncomeYes * PYesStudentYes)
print(PYesBuyA)

0.5686274509803921


In [24]:
#Probabilitas ClasYes dengan StudentYes, IncomeLow dan AgeSedang
PYesBuyB = ((PSedangYes * PLowIncomeYes * PYesStudentYes) * PYes) / (PSedangYes * PLowIncomeYes * PYesStudentYes)
print(PYesBuyB)

0.5686274509803921


In [25]:
#Probabilitas ClasYes dengan StudentYes, IncomeLow dan AgeBesar
PYesBuyC = ((PBesarYes * PLowIncomeYes * PYesStudentYes) * PYes) / (PBesarYes * PLowIncomeYes * PYesStudentYes)
print(PYesBuyC)

0.5686274509803921


In [26]:
#P(A) = Clas(buy_computer)  = NoBuy = PNo

#Kondisi B Dimana
#Age = Sedang
#Income = Low
#Student = Yes
#P(B) = PSedang, PLow, PYesStudent

#P(B)1 = PSedang
#P(B)2 = PLow
#P(B)3 = PYesDStudent

#P(B|A)1 = (PSedang | NoBuy) = PSedangNo
#P(B|A)2 = (PLow | NoBuy) = PLowIncomeNo
#P(B|A)3 = (PYesStudent | NoBuy) = PYesStudentNo


#Yang ditanyakan P(A|B) ?
#Rumus
#P(A|B) = P(B|A) * P(A) / P(B)
#P(NoBuy) = ((PSedangNo * PLowIncomeNo * PYesStudentNo) * PNo) / (PSedangNo * PLowIncomeNo * PYesStudentNo)

In [27]:
PNoBuy = ((PSedangNo * PLowIncomeNo * PYesStudentNo) * PNo) / (PSedangNo * PLowIncomeNo * PYesStudentNo)
print(PNoBuy)

0.43137254901960786


### Coclusion II
* Probability of Clas(buy_computer) = 0.5686274509803921
* Probability of no Clas(buy_computer) = 0.43137254901960786


## Result 

In [28]:
PSum = PYesBuyB + PNoBuy
print(PSum)

1.0


In [29]:
PBuy1 = (PYesBuyB / PSum) * 100
print(PBuy1)

56.86274509803921


In [30]:
PNoBuy1 = (PNoBuy / PSum) * 100
print(PNoBuy1)

43.13725490196079


## Jawaban
* Berapa persen tingkat pembelelian komputer dari Student = Yes dengan tingkat income = Low ?
* Jawabannya adalah 56% (didapat dari PBuy1)